crates/shared/src/sources/balancer_v2/pools.rs

```diff
@@ -24,6 +24,16 @@ pub trait FactoryIndexing {
     /// be retrieved once. This data will be passed in when fetching the current
     /// pool state via `fetch_pool`.
     type PoolInfo: PoolIndexing;
+
+    /// Augments the specified common pool info for this factory.
+    ///
+    /// This allows pool factories like the `WeightedPoolFactory` to add
+    /// `weights` to the common pool info, since these are declared as
+    /// `immuatble` in the smart contract and thus can never change and don't
+    /// need to be re-fetched.
+    ///
+    /// Returns an error if fetching the augmented pool data fails.
```

Comment:
nanonit: A check to verify that the given pool id in in fact of type `Self::PoolInfo` is not mandatory.